src/systemd/sd-dhcp6-client.h

```diff
@@ -184,8 +184,13 @@ enum {
         SD_DHCP6_OPTION_SLAP_QUAD                  = 140, /* RFC 8948 */
         SD_DHCP6_OPTION_V6_DOTS_RI                 = 141, /* RFC 8973 */
         SD_DHCP6_OPTION_V6_DOTS_ADDRESS            = 142, /* RFC 8973 */
-        SD_DHCP6_OPTION_IPV6_ADDRESS_ANDSF         = 143 /* RFC 6153 */
+        SD_DHCP6_OPTION_IPV6_ADDRESS_ANDSF         = 143, /* RFC 6153 */
         /* option codes 144-65535 are unassigned */
+        /* DHCPv6 private options. Assuming private options base as 224
+         * derived from DHCPv4 and end as 65535 */
```

Comment:
Hm??? Is it true?? Could you provide references?

Reply:
> Hm??? Is it true?? Could you provide references?

There are no private DHCPv6 options. There was apparently an attempt to introduce something like that in https://datatracker.ietf.org/doc/html/draft-volz-dhc-dhcpv6-site-options-00 but it didn't go anywhere. As far as I understand the idea and the term is used here to make the DHCPv6 client consistent with the DHCPv4 client or something like that. (personally I don't think it makes sense)

Reply:
Agreed, this is not specifically mentioned in RFC regarding DHCPv6 private options. I assumed the starting range for these options as 224 based on DHCPv4 as the range 148 to 65535 is unassigned for DHCPv6.

Reply:
I think I get that. What I don't understand is why there should be an API exposing options that aren't supposed to be used in the first place and why they are called private even though there aren't private (and there is no such thing as a private DHCPv6 option). It could be I'm overthinking this but I don't understand why this API should exist.

Reply:
I am totally inline with your comment that these are not really private options as per DHCPv6 RFC, but these are vendor/organization specific options passed by the DHCP server. Currently there is no way to query these options via dbus API and which is the reason I added these to the json output.

Reply:
> these are vendor/organization specific options passed by the DHCP server

That's what the Vendor-specific Information Option is for: https://datatracker.ietf.org/doc/html/rfc8415#section-21.17